contracts/libs/0.8.x/BytecodeStorage.sol

```diff
@@ -0,0 +1,257 @@
+ *
+ * @dev Compared to the above two rerferenced libraries, this contracts-as-storage implementation makes a few
+ *      notably different design decisions:
+ *      - uses the `string` data type for input/output on reads, rather than speaking in bytes directly
+ *      - exposes "delete" functionality, allowing no-longer-used storage to be purged from chain state
+ *      - stores the "writer" address (library user) in the deployed contract bytes, which is useful for both:
+ *         a) providing necessary information for safe deletion; and
+ *         b) allowing this to be introspected on-chain
+ *      Also, given that much of this library is written in assembly, this library makes use of a slightly
+ *      different convention (when compared to the rest of the Art Blocks smart contract repo) around
+ *      pre-defining return values in order to simplify need to directly memory manage these return values.
+ */
+library BytecodeStorage {
+    //---------------------------------------------------------------------------------------------------------------//
+    // Offset Amount | Offset Aggregate | Description                                                                //
+    //---------------------------------------------------------------------------------------------------------------//
+    // 53            | 53               | allow contract to be `selfdestruct`-able via gated-cleanup-logic           //
+    // 20            | 73               | reserve 20 bytes for storing deploying-contract's address                  //
+    //---------------------------------------------------------------------------------------------------------------//
+    // define the offset for where the "logic bytes" end, and the "data bytes" begin
+    uint256 internal constant DATA_OFFSET = 73;
+
+    /*//////////////////////////////////////////////////////////////
+                           WRITE LOGIC
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Write a string to contract bytecode
+     * @param _data string to be written to contract
+     * @return address_ address of deployed contract with bytecode containing concat(gated-cleanup-logic, data)
+     */
+    function writeToBytecode(string memory _data)
+        internal
+        returns (address address_)
+    {
+        // prefix bytecode with
+        bytes memory creationCode = abi.encodePacked(
+            //---------------------------------------------------------------------------------------------------------------//
+            // Opcode  | Opcode + Arguments  | Description  | Stack View                                                     //
+            //---------------------------------------------------------------------------------------------------------------//
+            // (0) creation code returns all code in the contract except for the first 11 (0B in hex) bytes
```

Comment:
I was doing some digging, and I don't fully understand why this is offset by 11 bytes specifically. Why 11?

Reply:
These 11 bytes are excluded as we don't want to include the contract creation code itself in the deployed contract data, which is the length of this contract creation code (11 bytes)–will add comment to clarify this inline.